lib/std/crypto/rsa.zig

```diff
@@ -0,0 +1,873 @@
+    }
+};
+
+/// Deprecated.
+///
+/// Signature Scheme with Appendix v1.5 (RSASSA-PKCS1-v1_5)
+///
+/// This standard has been superceded by PSS which is formally proven secure
+/// and has fewer footguns.
+pub fn PKCS1v1_5(comptime Hash: type) type {
+    return struct {
+        const PkcsT = @This();
+        pub const Signature = struct {
+            bytes: []const u8,
+
+            const Self = @This();
+
+            pub fn verifier(self: Self, public_key: PublicKey) !Verifier {
+                return Verifier.init(self, public_key);
+            }
+
+            pub fn verify(self: Self, msg: []const u8, public_key: PublicKey) !void {
+                var st = Verifier.init(self, public_key);
+                st.update(msg);
+                return st.verify();
+            }
+        };
+
+        pub const Signer = struct {
+            h: Hash,
+            key_pair: KeyPair,
+
+            fn init(key_pair: KeyPair) Signer {
+                return .{
+                    .h = Hash.init(.{}),
+                    .key_pair = key_pair,
+                };
+            }
+
+            pub fn update(self: *Signer, data: []const u8) void {
+                self.h.update(data);
+            }
+
+            pub fn finalize(self: *Signer, out: []u8) !PkcsT.Signature {
+                const k = self.key_pair.public.modulus.byteLen();
+                if (out.len < k) return error.BufferTooSmall;
+
+                var hash: [Hash.digest_length]u8 = undefined;
+                self.h.final(&hash);
+
+                const em = try emsaEncode(hash, out[0..k]);
+                try self.key_pair.encrypt(em, em);
+                return .{ .bytes = em };
+            }
+        };
+
+        pub const Verifier = struct {
+            h: Hash,
+            sig: PkcsT.Signature,
+            public_key: PublicKey,
+
+            fn init(sig: PkcsT.Signature, public_key: PublicKey) Verifier {
+                return Verifier{
+                    .h = Hash.init(.{}),
+                    .sig = sig,
+                    .public_key = public_key,
+                };
+            }
+
+            pub fn update(self: *Verifier, data: []const u8) void {
+                self.h.update(data);
+            }
+
+            pub fn verify(self: *Verifier) !void {
+                const pk = self.public_key;
+                const s = try Fe.fromBytes(pk.modulus, self.sig.bytes, .big);
+                const emm = try pk.modulus.powPublic(s, pk.public_exponent);
+
+                var em_buf: [max_modulus_len]u8 = undefined;
+                const em = em_buf[0..pk.modulus.byteLen()];
+                try emm.toBytes(em, .big);
+
+                var hash: [Hash.digest_length]u8 = undefined;
+                self.h.final(&hash);
+
+                // TODO: compare hash values instead of emsa values
+                const expected = try emsaEncode(hash, em);
+
+                if (!std.mem.eql(u8, expected, em)) return error.Inconsistent;
```

Comment:
Not sure if this needs to be constant time or not. It's not mentioned anywhere, so I guessed it's okay to use the fast version.